post_office/signals.py

```diff
@@ -0,0 +1,13 @@
+
+from django.db.models.signals import post_init
+from django.dispatch import receiver
+
+from .models import Email
+from .fields import CommaSeparatedEmailField
+
+
+@receiver(post_init, sender=Email)
+def on_email_post_init(sender, instance, **kwargs):
```

Comment:
The problem is that `Email.__init__` does not execute `CommaSeparatedEmailField.__init__`. So the following unit test never runs CommaSeparateEmailField's processing and therefore fails:

```python
 def test_models_repr(self):
        self.assertEqual(repr(Email(to='test@example.com')), "<Email: ['test@example.com']>")
```

The output of repr() is simply `<Email: 'test@example.com'>`. The preprocessing/validation must be done in a form instead and `Email.__init__` simply expects the one and correct `['test@example.com']` list value from its caller.

The only way to make that unit test pass is to either override `Email.__init__` (which Django does not recommend) or to use signals as I did above. Alternatively, maybe it's best to consider the test as broken because it tests something that Django doesn't do. 

Consider the following example:

```python
from django.db.models import Model, IntegerField
class MyModel(Model):
    number = IntegerField()
    def __repr__(self):
        return "MyModel(number=%s)" % repr(self.number)

my_instance = MyModel(number="obviously_not_an_integer")
print(repr(my_instance))
```

Which will simply output `MyModel(number='obviously_not_an_integer')` even though my_instance.number clearly conflicts with the IntegerField for the same reason.